department-of-veterans-affairs/caseflow
# Add explain endpoint for appeals for System Admins only

Innovation project: [Appeal Narrative](https://dsva.slack.com/archives/C022T6KSE6S/p1622571159006000)

This first PR provides data (and data dumps) to be potentially shown to a System Admin user. Subsequent PRs will replace the frontend with a more non-engineer friendly UI.

### Description

Add `explain` endpoint for appeals with placeholder files. 
The endpoint only works for System Admins (i.e., `user.admin? => true`).

Shows task tree and intake rendering in the browser.

Changes copied from draft PR #16039.

### Acceptance Criteria
- [ ] `explain` endpoint shows data about the appeal

### Testing Plan
1. In dev Rails console, list System Admin users: `Functions.list_all`
2. In Caseflow, log in as a System Admin (e.g., `BVASCASPER1`)
3. Go to their queue: http://localhost:3000/queue and click on a case to open Case Details. Try with AMA and legacy cases.
4. Change the url from `queue` to `explain`, like this: `http://localhost:3000/explain/appeals/31927be1-8447-44c4-ad66-8b73ae8bf401`
5. Click on the blue 'Help' text
   - Append `?show_pii`to the URL like this: `http://localhost:3000/explain/appeals/31927be1-8447-44c4-ad66-8b73ae8bf401?show_pii` or click on the link
   - Customize task tree columns like this: `http://localhost:3000/explain/appeals/31927be1-8447-44c4-ad66-8b73ae8bf401?fields=id,status,updated_at,CLO_DATE` or click on the link for an example
   - Click on the `json` link to show the JSON file contents, which can then be examined for PII and then saved locally for importing.

Repeat the above logged in as a non-admin. Should see an error:
```
{"errors":[{"status":403,"title":"Additional access needed","detail":"Additional access needed"}]}
```

You can also add `binding.pry` in `explain_spec.rb` and examine the page.

### User Facing Changes
![image](https://user-images.githubusercontent.com/55255674/112224053-c3ea6200-8bf8-11eb-903b-2ee5405a12a7.png)
![image](https://user-images.githubusercontent.com/55255674/112227334-7d4b3680-8bfd-11eb-8ed5-d9f1b1a70df3.png)

### Code Documentation Updates
